AllenInstitute/AllenSDK
# 2167 cloudcache from s3

This PR is intended to address #2167

Briefly, this set of changes is intended to support instantiating a VisualBehaviorOphysProjectCache from an S3 bucket that is mounted as a local file system. This is necessary for the Summer Workshop for the Dynamic Brain where AllenSDK will be run in Sagemaker IPython notebooks which have visual behavior ophys data accessibly via a mounted S3 bucket.

Validation:
- [x] I have already tried the following and it appears to be working, but having others try it out would also be good.

You can test out this new functionality by:
1. Install s3fs (https://github.com/s3fs-fuse/s3fs-fuse)
2. Make directory you want to act as S3 mount point (e.g. `mkdir /remote_data`)
3. Run `s3fs visual-behavior-ophys-data /remote_data/ -o default_acl="public-read" -o complement_stat,uid=$(id -u),gid=$(id -g),umask=222,public_bucket="1",endpoint="us-west-2"`
4. Run [example notebooks](https://allensdk.readthedocs.io/en/latest/visual_behavior_optical_physiology.html) but when instantiating the project cache use:

```
data_storage_directory = Path("/remote_data/")

cache = VisualBehaviorOphysProjectCache.from_local_cache(cache_dir=data_storage_directory, use_static_cache=True)
```
5. Don't forget to `umount` the S3 bucket
